core/src/main/java/io/opencensus/stats/export/StatsExporter.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2017, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.stats.export;
+
+import io.opencensus.stats.View;
+import io.opencensus.stats.ViewData;
+import java.util.Collection;
+import javax.annotation.concurrent.ThreadSafe;
+
+/** A service that is used by the library to export {@code ViewData}. */
+@ThreadSafe
+public abstract class StatsExporter {
+
+  private static final NoopStatsExporter NOOP_STATS_EXPORTER = new NoopStatsExporter();
+
+  /**
+   * Returns the no-op implementation of the {@code StatsExporter}.
+   *
+   * @return the no-op implementation of the {@code StatsExporter}.
+   */
+  public static NoopStatsExporter getNoopStatsExporter() {
+    return NOOP_STATS_EXPORTER;
+  }
+
+  /**
+   * Registers a new service handler that is used by the library to export {@code ViewData}.
+   *
+   * @param name the name of the service handler. Must be unique for each service.
+   * @param handler the service handler that is called for each ended sampled span.
+   */
+  public abstract void registerHandler(String name, Handler handler);
```

Comment:
Why are there two methods for registering handlers?  It looks like a handler can be registered with `ViewManager.registerView(View view, List<? extends Handler> handlers)` or `StatsExporter.registerHandler(String name, Handler handler)`.

Reply:
`StatsExporter.registerHandler(String name, Handler handler)` is used to register handlers to StatsExporter (so that stats can be pushed to them), `ViewManager.registerView(View view, List<? extends Handler> handlers)` is for registering views against handlers. A handler must be registered on StatsExporter first, and then views can be registered for the handler.

Reply:
An alternative is to completely remove StatsExporter and ExportComponent from API. StatsExporter will just be a worker thread in the implementation, and once user calls ViewManager.registerView(View view, List<? extends Handler> handlers), stats for the given view will be automatically exported to the handlers.

This makes our API simpler to use, but might also make it hard to mock and test.

Reply:
The alternative sounds simpler, and I like that it would move code out of the API.  How would it affect mocking or testing?  Would we lose any other functionality by moving it out of the API library?

Reply:
Right, I just realized moving this to implementation won't affect the functionality or mocking. I'll go ahead with the alternative approach to minimize API changes.